src/vector/_backends/numpy_.py

```diff
@@ -709,16 +709,14 @@ def _wrap_result(
             and issubclass(returns[0], Azimuthal)
         ):
             result = _toarrays(result)
-            dtype = []
-            i = 0
-            for name in _coordinate_class_to_names[returns[0]]:
-                dtype.append((name, result[i].dtype))
-                i += 1
+            dtype = [
+                (name, result[i].dtype)
+                for i, name in enumerate(_coordinate_class_to_names[returns[0]])
+            ]
+
             out = numpy.empty(_shape_of(result), dtype=dtype)
-            i = 0
-            for name in _coordinate_class_to_names[returns[0]]:
+            for i, name in enumerate(_coordinate_class_to_names[returns[0]]):
                 out[name] = result[i]
-                i += 1
```

Comment:
`i` is still used, it's just used via enumerate, instead of manually setting and incrementing it.

Reply:
`enumerate` is the correct Python syntax for this. Setting a variable and manually incrementing it is error prone if you forget to increment, maybe due to branching, and a reader can see the enumerate and immediately know this is a loop counter variable, and not a psuedo counter number (say one that increments more or fewer times than the loop).

Reply:
In `_wrap_results` functions like this, not this one, the `i` needs to be shared across multiple `for` loops. That's why it's written this way, so that the similarity of this function to the others will be more apparent when they're compared side by side.

The point is that you're not always asking, "What does this function do?" Sometimes you're asking, "How is this function different from this other function?" (which has the same name and belongs to VectorNumpy3D or VectorNumpy4D, rather than VectorNumpy2D). Using the same idioms in all of them, even if they're not the simplest idioms for the simplest case, makes it easier to answer the latter question.

Except for the one that failed CI, all of these are stylistic things and I'm hereby giving you final say in what goes into the library. I'm just pointing out the human readability reasons why I wrote them this way, which automated style metrics can't quantify.

Reply:
Let me compare, I'm mostly looking at these in isolation, much like the automated checks. In general, though, the fact that this is _not_ shared is important; making it "appear" shared when it is not can make it harder to see what really is going on in the function. That highlights the difference when it is shared when it's written differently. The name matches, so you still get symmetry there.

If `i, v = enumerate(vs)` always means that is not shared between loops, and `i = 0`, `i+=0` is only used when it is shared between loops, it highlights this so that the reader immediately knows to look for the sharing, because something is different. The fact that it's shared is very important information for reading and understanding what is happening!

Reply:
Here's an example of the case that we want to compare the above to:

https://github.com/scikit-hep/vector/blob/a42725b8eddf240e518bc2aa0f207dc2fcf08b42/src/vector/_backends/numpy_.py#L1315-L1347

When there are multiple kinds of coordinates, azimuthal, longitudinal, and maybe also temporal, then the counter has to start not at zero after the first one. I agree 100% that `enumerate` is the right way to do it when you have only one, but when you have two or three, using `enumerate` would make it less clear because you'd have to add an offset to the counter `enumerate` gives you. This way, the final value of `i` in one `for` loop is the initial value of `i` in the next `for` loop, which is exactly what _this_ application needs.

Again, I think you should have final say about how the code in this library is written, but this is _why_. Now you can choose whether the single-function clarity is more important than cross-function clarity.

Reply:
Remember, a reader is _not_ seeing this as a whole at first, they are seeing just the part they are reading at first. So if they see this, they will have to carefully check to see why `i` is being set this way, and there will be no reason, it's not shared here. When they finally come on one that it matters with, they no longer are checking explicitly, and are just assuming the author liked writing loops this way, and may completely miss there's a very important difference now and `i` is actually shared between loops.

Reply:
If enumerate let you set a start value, I'd be tempted. :)

I think I like the enumerate when the loops are not shared. I think it makes those loops cleaner, and using the same name (`i`) provides enough symmetry. To my eye, I can read this much more easily with the enumerate on the ones without variable sharing, and clear sharing construct (`i=0`) on the ones that do. Though if you still feel even mildly that you'd like the original better, happy to revert.

Reply:
It would be awesome if `enumerate` let you set a start value.

Reply:
It _almost_ did. But Guido thought it would be confusing with slices. See https://www.python.org/dev/peps/pep-0279/ under Note D. (old PEPs have fewer anchors)